src/kernel/src/fs/mod.rs

```diff
@@ -272,6 +273,27 @@ impl Fs {
         Ok(written.into())
     }
 
+    fn sys_writev(self: &Arc<Self>, i: &SysIn) -> Result<SysOut, SysErr> {
+        let fd: i32 = i.args[0].try_into().unwrap();
+        let ptr: *const libc::iovec = i.args[1].into();
+        let len: usize = i.args[2].try_into().unwrap();
+
+        let td = VThread::current().unwrap();
+        let file = td.proc().files().get(fd).ok_or(SysErr::Raw(EBADF))?;
+        let mut written = 0;
+        for i in 0..len - 1 {
+            let buf: &[u8] = unsafe {
+                std::slice::from_raw_parts(
```

Comment:
Also, it doesn't really make sense to create an iovec(which is basically a slice) and turn it back into a slice again. 
But this will have to be changed during the refactor.